src/libraries/System.Runtime.Serialization.BinaryFormat/src/PayloadReader.cs

```diff
@@ -0,0 +1,285 @@
+    /// <exception cref="ObjectDisposedException">The stream was closed.</exception>
+    /// <remarks><para>It does not modify the position of the stream.</para></remarks>
+    public static bool ContainsBinaryFormatterPayload(Stream stream)
+    {
+#if NETCOREAPP
+        ArgumentNullException.ThrowIfNull(stream);
+#else
+        if (stream is null) throw new ArgumentNullException(nameof(stream));
+#endif
+        long beginning = stream.Position;
+        if (stream.Length - beginning < SerializedStreamHeaderRecord.Size + 2)
+        {
+            return false;
+        }
+
+        try
+        {
+            int firstByte = stream.ReadByte();
+            if (firstByte != (byte)RecordType.SerializedStreamHeader)
+            {
+                return false;
+            }
+
+            stream.Seek(-1, SeekOrigin.End);
+            int lastByte = stream.ReadByte();
+            return lastByte == (byte)RecordType.MessageEnd;
+        }
+        finally
+        {
+            stream.Position = beginning;
+        }
+    }
+
+    /// <summary>
+    /// Reads the provided Binary Format payload.
+    /// </summary>
+    /// <param name="payload">The Binary Format payload.</param>
+    /// <param name="options">An object that describes optional <seealso cref="PayloadOptions"/> parameters to use.</param>
+    /// <param name="leaveOpen">True to leave the <paramref name="payload"/> payload open
+    /// after the reading is finished, otherwise, false.</param>
+    /// <returns>A <seealso cref="SerializationRecord"/> that represents the root object.
+    /// It can be either <seealso cref="PrimitiveTypeRecord{T}"/>,
+    /// a <seealso cref="ClassRecord"/> or an <seealso cref="ArrayRecord"/>.</returns>
+    /// <exception cref="ArgumentNullException">When <paramref name="payload"/> is null.</exception>
+    /// <exception cref="ArgumentException">The <paramref name="payload"/> payload does not support reading or is already closed.</exception>
+    /// <exception cref="SerializationException">When reading input from <paramref name="payload"/> encounters invalid Binary Format data.</exception>
+    /// <exception cref="DecoderFallbackException">When reading input from <paramref name="payload"/>
+    /// encounters invalid sequence of UTF8 characters.</exception>
+    public static SerializationRecord Read(Stream payload, PayloadOptions? options = default, bool leaveOpen = false)
+        => Read(payload, out _, options, leaveOpen);
+
+    /// <param name="payload">The Binary Format payload.</param>
+    /// <param name="recordMap">Record map</param>
+    /// <param name="options">An object that describes optional <seealso cref="PayloadOptions"/> parameters to use.</param>
+    /// <param name="leaveOpen">True to leave the <paramref name="payload"/> payload open
+    /// after the reading is finished, otherwise, false.</param>
+    /// <inheritdoc cref="Read(Stream, PayloadOptions?, bool)"/>
```

Comment:
It's weird to me to combine inheritdoc with docs.  I don't even know what that does.

```suggestion
    /// <inheritdoc cref="Read(Stream, PayloadOptions?, bool)"/>
```

Reply:
> I don't even know what that does.

It combines one with another and lets me reduce the comment duplication:

![image](https://github.com/dotnet/runtime/assets/6011991/18afddbd-b7a9-4d3a-ab24-b7d9628645a3)

I would prefer to keep it, but I don't have very strong preferences here.